token-account-based/chaincode-go/chaincode/token_contract.go

```diff
@@ -83,63 +86,75 @@ func (s *SmartContract) Mint(ctx contractapi.TransactionContextInterface, amount
 	return nil
 }
 
-// Transfer transfers tokens from client account to recipient account.
-// recipient account must be a valid clientID as returned by the ClientID() function.
-func (s *SmartContract) Transfer(ctx contractapi.TransactionContextInterface, recipient string, amount int) error {
+// _Transfer is a helper function that transfers tokens from the "from" address to the "to" address.
+// Dependant functions include Transfer and TransferFrom.
+func (s *SmartContract) _Transfer(ctx contractapi.TransactionContextInterface, from string, to string, value int) error {
```

Comment:
Ok, for Go we can just use lowercase function names to keep them private (unexported)
I still have the question for Javascript... I thought leading underscore was just a convention but not enforced by Javascript.

Reply:
For a pure javascript implementation, the _ is enforced by the contract layer.  So those functions are not callable. 
TypeScript & Java - this is governed by the annotations - if the function is annotated it's callable, otherwise it's not.

Go follows the lowercase syntax.